migration/migration/003-users.sql

```diff
@@ -65,14 +65,21 @@ CALL _prom_catalog.execute_everywhere('grant_prom_reader_prom_writer',$ee$
     $$;
 $ee$);
 
-CALL _prom_catalog.execute_everywhere('grant_all_roles_to_extowner',$ee$
+CALL _prom_catalog.execute_everywhere('grant_all_roles_to_extowner',
+format(
+$ee$
     DO $$
     BEGIN
-        GRANT prom_reader TO @extowner@ WITH ADMIN OPTION;
-        GRANT prom_writer TO @extowner@ WITH ADMIN OPTION;
-        GRANT prom_maintenance TO @extowner@ WITH ADMIN OPTION;
-        GRANT prom_modifier TO @extowner@ WITH ADMIN OPTION;
-        GRANT prom_admin TO @extowner@ WITH ADMIN OPTION;
+        GRANT prom_reader TO %1$s WITH ADMIN OPTION;
+        GRANT prom_writer TO %1$s WITH ADMIN OPTION;
+        GRANT prom_maintenance TO %1$s WITH ADMIN OPTION;
+        GRANT prom_modifier TO %1$s WITH ADMIN OPTION;
+        GRANT prom_admin TO %1$s WITH ADMIN OPTION;
     END
     $$;
-$ee$);
+$ee$, CASE
+        WHEN version() LIKE 'PostgreSQL 12%' THEN quote_ident(current_user)
```

Comment:
I think this might be "safer". At least, it "feels safer" to me.

```
CASE WHEN current_setting('server_version') like '12.%' THEN ...
```

Reply:
Will be a long time before PostgreSQL 120 is released, but might at least consider `12.%` instead of `12%`. Did version `12.0` include the `.0` in this value?

Reply:
Great catch, thank you!